docs/developers/basic/hashes.md

```diff
@@ -0,0 +1,293 @@
+        hide-results
+      >
+      </call-zome-fns>
+      <div style="display: flex; flex-direction: row; align-items: start;">
+        <entry-graph
+          .showFilter=${false}
+          .excludedEntryTypes=${["Agent"]}
+          style="flex: 1; margin-right: 20px; height: 500px;"
+        >
+        </entry-graph>
+        <entry-detail style="flex-basis: 600px; height: 500px;"> </entry-detail>
+      </div>
+    </holochain-playground-container>
+  `;
+};
+```
+
+First thing you might notice is that each hash has the same length. No matter how small or how big the input is, the hash is always the same length. If you don't believe it, try it. Try a really long greeting.
+The second thing you will find is that hashes do not look very similar. In fact, even the smallest difference in the input will give a very different hash. Again, try it out! Create an entry for *hello world* all lowercase, or add a comma or even a space.
+The hashes will be very different.
+Finally for the grand finale: add *Hello world* as an entry and then add it again. What do you notice?
+
+Nothing happens. And that is a feature, not a bug. Because the same input, always returns the same hash. So when you add the *Hello World* entry a second and a third time the Entry Graph detects it already has an entry with this hash. And why save it again, if you already have it. It is a smart way to store data.
+
+## Hash table
+
+When you combine all these properties of hashes you can do interesting stuff. If for every piece of data or content you have, you can calculate the hash value, then you could easily make a nice big list of all the hashes. No matter the size of the data you want to store, each piece will have it's own hash, which is unique but has the same size as all other hashes. And since all those hashes can be mapped to their content, you can use that list as an index. This type of list has a name, it's a **hash table**. The fancy name for this way of storing and retrieving your data this way is called **content addressable storage**. Because the content, well actually the hash of the content, is be used as a unique address for the content. It will really get interesting when we upgrade this hash table to a distributed hash table, a DHT.
+
+## Getting ready
+
+So I hope you see that hashes play a big role in how data is stored and retrieved. You will see them poping up when we talk about headers, elements, the DHT, validations. They are everywhere.  
+Enough talking. It is time to get ready for the exercise. 
+
+This time you are building a library zome. The zome will need 2 functions that can be called from the outside: `add_book` and `get_book`. You will add your favorite book and retrieve it from the library zome based on -you guessed it- the hash of its entry. In one of the next exercise we will see that there are better ways to retrieve your data: headers and elements. But it is good to know you can always find your entry again based on its entry hash.
```

Comment:
I wouldn't say "there are better ways to retrieve your data", more like "there are other ways to retrieve your data". Depending on the context, it's much better to get by entry hash, in others it's the opposite.